common/src/main/scala/com/gu/sfl/persistence/SavedArticlesPersistence.scala

```diff
@@ -32,8 +34,15 @@ class SavedArticlesPersistenceImpl(persistanceConfig: PersistenceConfig) extends
     val articles = mapper.readValue[List[SavedArticle]](dynamoSavedArticles.articles)
     SavedArticles(dynamoSavedArticles.version, articles)
   }
-
-  private val client = DynamoDbClient.create()
+  private val client = DynamoDbClient
+                        .builder()
+                        .credentialsProvider(
+                            AwsCredentialsProviderChain.of(EnvironmentVariableCredentialsProvider.create(),
+                              SystemPropertyCredentialsProvider.create(),
+                              ProfileCredentialsProvider.create("mobile"),
+                              InstanceProfileCredentialsProvider.create())
```

Comment:
I think for an AWS Lambda you only ever need to provide `EnvironmentVariableCredentialsProvider` (which is how a Lambda running in AWS infrastructure gets its AWS tokens) and `ProfileCredentialsProvider` (which is what our code uses when it's running locally on our laptops with credentials from Janus). An example of this configuration is in the [`google-search-indexing-observatory` AWS Lambda](https://github.com/guardian/google-search-indexing-observatory/blob/587cdf235f5536993962266a53ad7175d33cee3d/src/main/scala/ophan/google/indexing/observatory/AWS.scala#L15-L16).



```suggestion
                            AwsCredentialsProviderChain.of(EnvironmentVariableCredentialsProvider.create(),
                              ProfileCredentialsProvider.create("mobile"))
```

This change can also be applied to `SflDynamoDb.scala` below!